pupil_src/shared_modules/math_helper/transformations.py

```diff
@@ -1980,6 +1983,33 @@ def _import_module(name, package=None, warn=True, prefix="_py_", ignore="_"):
         return True
 
 
+def find_rigid_transform(A, B):
+    # we expect the shape to be of length 2
+    assert len(A.shape) == len(B.shape) == 2
+    assert A.shape[0] == B.shape[0]
+
+    centroid_A = np.mean(A, axis=0)
+    centroid_B = np.mean(B, axis=0)
+
+    # centre the points
+    A -= centroid_A
+    B -= centroid_B
+
+    # dot is matrix multiplication for array
+    H = A.T @ B
+    U, S, Vt = np.linalg.svd(H)
+    R = Vt.T @ U.T
+    # special reflection case
```

Comment:
Why is this case special and when does it occur? Should we throw an exception in this case? Forwarding this information without context to the end-user by logging it on the info level seems confusing/inappropriate to me.

Reply:
See above.